In the Kolab Cyrus Imap server mailboxes shared by other users are
named following the pattern:

  user/USERNAME/MAILBOX

Where USERNAME is the sharing users name _without_ the domain part.
Other IMAP servers however might specify the USERNAME including the
domain -- this is the case for example with some Dovecot setups.

The current code assumes the behavior of Cyrus and adds the users
domain to construct the fully qualified owner of an shared folder.

This Patch looks at the extracted USERNAME and if it already contains
an at-sign, if so no extra domain part is added.


diff -c a/a/horde-webmail/lib/Horde/Kolab/Storage/Folder.php b/horde-webmail/lib/Horde/Kolab/Storage/Folder.php
--- a/a/horde-webmail/lib/Horde/Kolab/Storage/Folder.php
+++ b/a/horde-webmail/lib/Horde/Kolab/Storage/Folder.php
@@ -494,9 +494,13 @@
             if (substr($matches[1], 0, 6) == 'INBOX/') {
                 $this->_owner = Auth::getAuth();
             } elseif (substr($matches[1], 0, 5) == 'user/') {
-                $domain = strstr(Auth::getAuth(), '@');
-                $user_domain = isset($matches[4]) ? $matches[4] : $domain;
-                $this->_owner = $matches[2] . $user_domain;
+                if (strpos($matches[2], '@') === FALSE) {
+                    $domain = strstr(Auth::getAuth(), '@');
+                    $user_domain = isset($matches[4]) ? $matches[4] : $domain;
+                    $this->_owner = $matches[2] . $user_domain;
+                } else {
+                    $this->_owner = $matches[2];
+                }
             } elseif ($matches[1] == 'shared.') {
                 $this->_owner =  'anonymous';
             }

Diff finished.  Wed Feb 18 17:55:20 2009
